### PR TITLE
Add an issue template

### PR DESCRIPTION
**Pull request by ranisalt**
Now that Github has added templates for issues and pull requests, it would be nice to add one and prevent people creating useless reports just for us to ask "what system?" "what compiler?" "did you checkout to the latest version?" and remember people with prehistoric compilers do not work.

Please check https://github.com/ranisalt/node-argon2/issues/new to see how it works

### CODE REVIEW

**Comment by WibbenZ** [comment]
This... would be perfect.
Would make even more sence on the support board haha :+1:

**Comment by ranisalt** [comment]
> Would make even more sence on the support board haha

Good one, forgot to warn people to check the support board before creating an issue

**Comment by Tarliton** [comment]
Is it good to add this?
Before creating an issue, please be sure to:
-try to install all the dependencies and compile again
I remember 2 or 3 issues open and the reason was that pugixml became a dependency

**Comment by ranisalt** [comment]
> Is it good to add this?

It is, but I think it's enough to ask for the dependency version. If one looks at "pugixml version" and does not has it installed, it will ring the bell.

**Comment by ranisalt** [comment]
Anything holding this back?

**Comment by marksamman** [comment]
I think it's asking too much which may discourage people from reporting issues. The biggest problem with issues is people using them for support matters, and sometimes writing comments that don't contribute to solving the issue. I don't recall any issue where MySQL version, PugiXML version or GMP version has mattered. There aren't many issues where the environment matters. It's also encouraging spam when it says to bump issues (as in just posting "bump", or "+1" or anything else that doesn't contribute to the issue).

**Comment by ranisalt** [comment]
> It's also encouraging spam when it says to bump issues (as in just posting "bump", or "+1" or anything else that doesn't contribute to the issue)

Hmmm I thought the opposite, it could prevent some duplicate issues which happens pretty a lot for older bugs (or sometimes not even a bug). I agree with the rest.

**Comment by ranisalt** [comment]
> It's also encouraging spam when it says to bump issues

Oh yeah, Github has just released an official feature to "+1" comments (and more): https://github.com/blog/2119-pull-request-and-issue-reactions I think on including it instead.

**Comment by marksamman** [comment]
That's a step in the right direction for GitHub, should make issues less noisy. As for this issue, I think something simpler, like this, would be better:
```
By creating this issue, I confirm that:
- [ ] This is a bug in the software that resides in this repository, and not a support matter (use https://otland.net/forums/support.16/ for support)
- [ ] I have not made any changes to the source code

### Steps to reproduce (include any configuration/script that is required to reproduce the issue)
1.
2.
3.

### Expected behavior
Tell us what should happen

### Actual behavior
Tell us what happened instead

### Environment
If the issue is environment specific (e.g. compiling errors), please include the name and version of the operating system and compiler you are using
```

File: ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+Welcome to the issues section if it's your first time!
+
+### Before creating an issue, please ensure:
+- [ ] This is a bug in the software that resides in this repository, and not a
+      support matter (use https://otland.net/forums/support.16/ for support)
+- [ ] You have not made changes to the source code related to the problem
```
Comment (marksamman) [inline ISSUE_TEMPLATE.md]: I'd rather be stricter about this, or at least reword it to say something like:
"The issue is reproducible without changes to the source code"

File: ISSUE_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+- [ ] Search for a similar issue, add your reaction there instead :tada:
```
Comment (marksamman) [inline ISSUE_TEMPLATE.md]: This depends on the situation, and I prefer to not have it in the template. Sometimes the person who would open a duplicate issue may complement the original issue with additional information, and I'd rather read their comment than just see their reaction even though the issue will be closed and marked as duplicate.

Comment (ranisalt) [inline ISSUE_TEMPLATE.md]: Better not to have or to rewrite to `add your reaction and complement information`?

Comment (marksamman) [inline ISSUE_TEMPLATE.md]: It's better to remove it altogether, I'd like to keep the template simple.

File: ISSUE_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
+Welcome to the issues section if it's your first time!
```
Comment (marksamman) [inline ISSUE_TEMPLATE.md]: This should be commented out.

File: ISSUE_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+      support matter (use https://otland.net/forums/support.16/ for support)
+- [ ] This issue is reproducible without changes to the code in this repository
+
+### Steps to reproduce (include any configuration/script required to reproduce)
+1.
+2.
+3.
+
+### Expected behaviour
+Tell us what should happen
```
Comment (marksamman) [inline ISSUE_TEMPLATE.md]: This should also be commented out.

File: ISSUE_TEMPLATE.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+### Actual behaviour
+Tell us what happens instead
```
Comment (marksamman) [inline ISSUE_TEMPLATE.md]: This should also be commented out.

File: ISSUE_TEMPLATE.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+### Environment
+If the issue is environment specific (e.g. compiling errors), please include the
```
Comment (marksamman) [inline ISSUE_TEMPLATE.md]: This should also be commented out.

File: ISSUE_TEMPLATE.md (hunk 3 above)
Comment (ranisalt) [inline ISSUE_TEMPLATE.md]: You mean, comment so that one creating an issue do not post this text accidentally?

Comment (marksamman) [inline ISSUE_TEMPLATE.md]: Yes. By wrapping it in ""

Comment (ranisalt) [inline ISSUE_TEMPLATE.md]: Good. HTML comments are prettier than Markdown comments.

**Comment by marksamman** [comment]
Thanks!
